### ISSUE TITLE
Implement checker for challenge caches

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#4549]
Challenge caches require a logging condition which most of the cases can not be checked by or in c:geo itself.
Example: 
http://coord.info/GC3RG33
This cache requires that the user has found 12 earth caches in one calendar day.
The Project-GC website offers checkers for those kinds of caches..in this example: http://project-gc.com/Challenges/GC3RG33/96

It would be a nice usecase if a c:geo user could trigger a check for the challenge from c:geo by using an API to project-gc.com.

Implementing this could be a good starting point for implementation of other features (e.g. complex cache searches, statistic views, etc.) with project-gc.com.

**Comment by Lineflyer** [comment]
Seems this might get very interesting now, that gc.com requires a web-based checker for all new challenge caches after end of the moratorium (refer to latest news).

Mabye the team of project-gc can establish a simple API where we submit the geocode and the username of the current logged in user and get back a "Yes"/"No"/"No checker" to find out if the user fullfills the logging conditions.
A menu for triggering the check yould be offered either if the cache name contains "challenge" or the description already contains a link to a project-gc checker. (Maybe those cache also get an own type...not yet known).

**Comment by Bananeweizen** [comment]
I like this. I would make it depend on the link to the checker for starters. However, you should probably first ask ganja, what he thinks of this. As a rough prototype, we could basically invoke the page that is normally invoked when you click the "Execute checker" button, and parse the result string "I did not qualify..." or similar.

**Comment by Lineflyer** [comment]
@magma1447 Just pinging you as @Bananeweizen used your old username.

**Comment by jakedot** [comment]
@Lineflyer @magma1447 are you guys still thinking about this (nice idea!) or can this issue be closed?

**Comment by moving-bits** [comment]
We already have a function in c:geo's cache detail page, which calls the project-gc geochecker:

![image](https://user-images.githubusercontent.com/3754370/75097828-6330c600-55af-11ea-82d3-6b49661df1db.png)

IMHO that's a working solution. What do you think?

(Alternatively, instead of using a separate menu entry, the same function could be combined with the "open geochecker" menu entry, or is there a specific reason why we want to have to separate menu entries?)

**Comment by jakedot** [comment]
@moving-bits Yes, the current behaviour is fine.

So, the purpose of this issue would currently be to think about making it even better, like automatically opening the log view with the result of the checker.

**Comment by jakedot** [comment]
> (Alternatively, instead of using a separate menu entry, the same function could be combined with the "open geochecker" menu entry, or is there a specific reason why we want to have to separate menu entries?)

Where is that "open geochecker" menu entry? At least by a perfunctory glance at the listing menu I could not see it.

**Comment by moving-bits** [comment]
> @moving-bits Where is that "open geochecker" menu entry? At least by a perfunctory glance at the listing menu I could not see it.

The "open geochecker" menu entry is only shown if there is a geochecker detected. For that, the listing text is searched for a couple of URL fragments.

**Comment by jakedot** [comment]
> > @moving-bits Where is that "open geochecker" menu entry? At least by a perfunctory glance at the listing menu I could not see it.
> 
> The "open geochecker" menu entry is only shown if there is a geochecker detected. For that, the listing text is searched for a couple of URL fragments.

Oh yeah makes sense to combine the two then, but can we make it so that it is only a submenu when there is a detected checker? Would it be possible if there are multiple checkers to list them by title or ID under the search entry in a submenu, and if there are no defections just display the search entry instead of the sub menu?

**Comment by magma1447** [comment]
I have obviously missed answering this post in the first pass, I did read it though.

I don't mind the links, quite the opposite, I appreciate them.

Regarding API. It was planned to make an official Project-GC-API last summer, but due to personal health, it didn't happen. Right now it feels like it's the second highest priority within development though, so hopefully I can begin on it in a few weeks.

It's not decided what the API will contain, so it's very open for suggestions. Here is a small list of thoughts that I have in mind right now.
* Most API methods will be usable by Project-GC's paying members only.
* Run a Challenge checker and receive a binary answer (actually, true, false or error). No additional information.
* Get a list of all Challenges, probably with Checker-links.
* Get a list of Virtual GPS's from user.
* Get content from Virtual-GPS (GC-codes and notes).
* No geocache metadata (D/T, size, corrected, cache notes, description ....) will be provided, due to licensing. The additional data has to be fetched by the partnering app.
* Each app/partner will have its own license key. It will be used in combination with a user token.

The idea is to setup a good framework which can easily be extended. While I appreciate thoughts on what methods should exist, it can wait. I don't have such notes yet and don't plan to write it down. But once there is a beta of the system, I will start collecting ideas.

**Comment by Lineflyer** [comment]
> (Alternatively, instead of using a separate menu entry, the same function could be combined with the "open geochecker" menu entry, or is there a specific reason why we want to have to separate menu entries?)

As per "normal" definition challenge caches are mysteries but the box is located at the listing coordinates. Thus I assume most challenge caches will not use a geochecker, which makes this discussion rather meaningless :)

**Comment by Lineflyer** [comment]
@magma1447 The API ideas sound good so far. Would you consider API challenge check to be only available paying PGC members?

**Comment by magma1447** [comment]
@Lineflyer Yes, today we have two income sources. Ads and memberships. Those who don't pay a membership at least bring in some ad-money, even though it's almost nothing. By moving them to a third party site/app via an api we would remove that small income as well.

Challenge checkers is also a way for us to bring new users into the web, so that they can discover an interest for it. By moving those users to the API we remove that incentive. For paying members it's not a big issue, because they already know about the site and find it worth it. We will gladly provide them with better alternatives.

Maintaining the data for the Challenge checkers are expensive. We are not getting paid by HQ to host these, but we pay (a lot) to get access to the data. This cost must be covered or we would have to shut down completely. If Geocachers wants the more convenient way of using our API, they will have to become paid members.

**Comment by wolverine007** [comment]
@magma1447 @Lineflyer 

long long time ago I can still remember how that "feature" used to make me smile... 


... so any news related to the announced API?

Maybe we could only offer it to Project-GC members with an active membership while the others are redirected to the website and still see ads to cover your cost? 

Cheers Wolv
